data/core/init.lua

```diff
@@ -329,6 +321,23 @@ local function scan_project_folder(index)
   else
     core.dir_rescan_add_job(dir, ".")
   end
+  if dir.name == core.project_dir then
+    core.project_files = dir.files
+  end
+end
+
+function core.open_folder_project(dir_path_abs)
+  if core.set_project_dir(dir_path_abs, core.on_quit_project) then
+    core.root_view:close_all_docviews()
+    update_recents_project("add", dir_path_abs)
+    if not core.load_project_module() then
+      command.perform("core:open-log")
+    end
+    for i = 1, #core.project_directories do
```

Comment:
Is this needed? `core.set_project_dir` will set it to a single one.

Reply:
If you mean the general function; I actually just moved it around, it was there before; just higher, but needed to be moved down to call other local functions. If you mean the for loop that I added to scan project directories; yes, it does need to be there, otherwise, on project change, we get a segfault (though perhaps this would be solved by your recommendation to change StatusView?)

Reply:
So, this wasn't fixed by the StatusView change. On project change, this is required to be here; many plugins seem to depend on this being the case.

Reply:
What I meant is that `core.set_project_dir` sets a single `core.project_directories`, so the loop was unnecessary, and just calling `scan_project_folder(1)` would have been fine.

Reply:
Ah, I see what you mean. 

However, the workspace plugin runs after `on_enter_project` and potentially adds more directories; so we should probably add with the appropriate index to `add_project_directory` instead. That'd handle that case, and the case where we add a directory with the command.

Reply:
Oh, lol. That was your line I deleted. Derp.